### PR TITLE
Add AOT benchmarks.

### PR DESCRIPTION
**Pull request by azteca1998**
# Add AOT benchmarks

## Description

Description of the pull request changes and motivation.

## Checklist
- [ ] Linked to Github Issue
- [ ] Unit tests added
- [ ] Integration tests added.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.

### CODE REVIEW

File: benches/benches.rs [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+        c.bench_function("Cached AOT factorial_2M (direct invoke)", |b| {
+            b.iter(|| aot_factorial_fn(u128::MAX));
+        });
+        c.bench_function("Cached AOT fib_2M (direct invoke)", |b| {
+            b.iter(|| aot_fibonacci_fn(u128::MAX));
+        });
+        c.bench_function("Cached AOT logistic_map (direct invoke)", |b| {
+            b.iter(|| aot_logistic_map_fn(u128::MAX));
+        });
```
Comment (Oppen) [inline benches/benches.rs:129]: Does this part need to be in the `unsafe` block?

File: benches/compile_time.rs [hunk 2]
```diff
@@ -44,7 +44,7 @@ pub fn bench_compile_time(c: &mut Criterion) {
                 b.iter(|| {
                     let native_context = NativeContext::new();
                     let module = native_context.compile(black_box(program)).unwrap();
-                    let object = module_to_object(module.module())
+                    let object = module_to_object(module.module(), OptLevel::Aggressive)
```
Comment (Oppen) [inline benches/compile_time.rs:47]: Is this one set to `Aggressive` to test the worst case? Maybe also try best case with `None` or `Default`?

**Review by Oppen — DISMISSED** [review]
Just two nits.

File: benches/compile_time.rs (hunk 2 above)
Comment (azteca1998) [reply benches/compile_time.rs:47]: Wouldn't aggressive be the best case?

File: benches/benches.rs (hunk 1 above)
Comment (azteca1998) [reply benches/benches.rs:129]: It was necessary on some of my first attempts, but not anymore. Fixed.

File: benches/compile_time.rs (hunk 2 above)
Comment (Oppen) [reply benches/compile_time.rs:47]: In compilation time? I expect aggressive optimizations to take longer.

**Review by igaray — APPROVED** [review]

**Review by edg-l — APPROVED** [review]

File: benches/compile_time.rs (hunk 2 above)
Comment (azteca1998) [reply benches/compile_time.rs:47]: Oh right, I didn't notice what was being benchmarked. Thanks.

**Review by igaray — DISMISSED** [review]

**Review by edg-l — APPROVED** [review]

**Review by igaray — APPROVED** [review]

**Pull request merged**